plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -965,6 +975,16 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         transaction = null;
     }
 
+    private List<DeleteFile> filterDeleteFiles(List<DeleteFile> deleteFiles, boolean skipped)
+    {
+        if (skipped) {
+            return deleteFiles.stream().filter(deleteFile -> deleteFile.content().equals(FileContent.POSITION_DELETES)).collect(toImmutableList());
+        }
+        else {
+            return deleteFiles;
+        }
+    }
```

Comment:
I think we want something like this:
```
private Set<DeleteFile> filterDeleteFiles(Set<DeleteFile> deleteFiles, boolean isWholeTableScan)
{
  if (isWholeTableScan) {
    return deleteFiles;
  }
  return deleteFiles.stream()
    .filter(deleteFile -> deleteFile.content().equals(POSITION_DELETES) || deleteFile.partition() != null)
    .collect(toImmutableSet());
}
```

If it's a position delete, we know we can remove it.
If it's an equality delete for a specific partition, we can remove it.

If it's a global equality delete, we can remove it iff the entire table was scanned.

Right?